vagrant/tutorial/Vagrantfile

```diff
@@ -4,10 +4,42 @@
 # Vagrantfile API/syntax version. Don't touch unless you know what you're doing!
 VAGRANTFILE_API_VERSION = "2"
 
+# This allows the VM to be restarted during provisioning, necessary after
+# updating the kernel
+unless Vagrant.has_plugin?("vagrant-reload")
+  raise "vagrant-reload plugin is not installed"
+end
+
+=begin
+  The following error may occur when running `vagrant up`, potentially because
+  of a problem with some VirtualBox versions.
+
+    # Failed to mount folders in Linux guest. This is usually because
+    # the "vboxsf" file system is not available. Please verify that
+    # the guest additions are properly installed in the guest and
+    # can work properly. The comm du yand attempted was:
+    #
+    # mount -t vboxsf -o uid=`id -u vagrant`,gid=`getent group vagrant | cut -d: -f3` vagrant /vagrant
+    # mount -t vboxsf -o uid=`id -u vagrant`,gid=`id -g vagrant` vagrant /vagrant
+    #
+    # The error output from the last command was:
+    #
+    # /sbin/mount.vboxsf: mounting failed with the error: No such device
```

Comment:
This seems a rather long comment to leave in the final Vagrantfile.
And I'm still not sure of the circumstances where this problem might occur.
Is it a MacOSX thing? If so, just say that and link to a GitHub issue.